netbox/dcim/forms/models.py

```diff
@@ -877,25 +877,33 @@ class PowerFeedForm(NetBoxModelForm):
             'site_id': '$site'
         }
     )
+    location = DynamicModelChoiceField(
+        queryset=Location.objects.all(),
+        required=False,
+        initial_params={
+            'racks': '$rack'
+        }
```

Comment:
Sorry, missed this earlier when we discussed: This should have `query_params` on it too, to limit the locations by the selected site:

```python
        query_params={
            'site_id': '$site'
        }
```